src/components/Common/Module/FieldPicker.vue

```diff
@@ -1,49 +1,126 @@
 <template>
-    <fieldset class="form-group">
+    <fieldset
+      class="form-group"
+      :disabled="disabled"
+    >
+      <b-form-group
+        :label="$t('block:recordList.fields')"
+        label-cols="2"
+        class="mt-3 h5"
+      >
+        <b-input-group
+          class="w-25"
+        >
+          <b-input
+            v-model.trim="query"
+            class="border-right-0 text-truncate"
+            type="search"
+            :placeholder="$t('module:searchPlaceholder')" />
+          <b-input-group-append>
+            <b-input-group-text class="text-primary bg-white border-left-0">
+              <font-awesome-icon
+                :icon="['fas', 'search']"
+              />
+            </b-input-group-text>
+          </b-input-group-append>
+        </b-input-group>
+      </b-form-group>
       <div class="fields d-flex">
```

Comment:
Lets used row and col structure for the two field areas (ensures proper padding and responsiveness)